README.md

```diff
@@ -1,86 +1,149 @@
+* [rocBLAS](https://github.com/ROCmSoftwarePlatform/rocBLAS) for running on the GPU
 * [HIP](https://github.com/ROCm-Developer-Tools/HIP) for running on the GPU
 * [Protobuf](https://github.com/google/protobuf) for reading [onnx](https://github.com/onnx/onnx) files
 * [Half](http://half.sourceforge.net/) - IEEE 754-based half-precision floating point library
 * [pybind11](https://pybind11.readthedocs.io/en/stable/) - for python bindings
+* [ONNX 1.7.0](https://github.com/onnx/onnx) and [Pytest](https://github.com/pytest-dev/pytest) for running the ONN backend
+  tests 
 
-## Installing the dependencies
+Note: 1) We have to use ONNX version 1.7.0 since changes in ONNX 1.8.0 is incompatible with our current implementation. 
+ONNX 1.7.0 can be installed as ```pip3 install onnx==1.7.0```, and updates to support ONNX version 1.8.0 will come soon. 2)
+MIOpen and rocBLAS can be installed as ```sudo apt update && sudo apt install -y miopen-hip rocblas```.
 
-Dependencies can be installed using the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
+### Installing the dependencies
+There are two alternative ways to install the above dependencies:
 
-To install rbuild:
-```
-pip install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
-```
+#### Run a shell script [build_prereqs.sh](./tools/build_prereqs.sh).
+
+(Note: 1. You need the sudo to install the above dependencies. 2. All dependencies are installed in default locations in
+the system and are accessible by all users.)
+
+#### Use the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
 
-To build dependencies along with MIGraphX
+To install rbuild (sudo may be needed.):
 ```
-rbuild build -d depend --cxx=/opt/rocm/llvm/bin/clang++
+pip3 install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
 ```
-This builds dependencies in the subdirectory named depend and then builds MIGraphX using these dependencies.
+and pip3 can be installed as `sudo apt update && sudo apt install -y python3-pip`
 
-## Building MIGraphX from source
+Dependencies are listed in files `requirements.txt` and `dev-requirements.txt` in the project diretory. To build the
+dependencies,
+* In ROCm3.3:
 
-## Configuring with cmake
-
-First create a build directory:
+```
+rbuild develop -d depend -B build --cxx=/opt/rocm/bin/hcc
+```
 
+* In ROCm3.7 and later releases:
 
 ```
-mkdir build; 
-cd build;
+rbuild develop -d depend -B build --cxx=/opt/rocm/llvm/clang++
 ```
 
-Next configure cmake. The hcc compiler is required to build the MIOpen backend:
+This builds dependencies in the folder "depend".
 
+Note that if rocBLAS and MIOpen are installed with the command ```sudo apt install -y rocblas miopen-hip``` as mentioned
+above, we can comment out the two lines 
+```
+ROCmSoftwarePlatform/rocBLAS@abd98a2b48b29326ebaef471630786a548622c06
+ROCmSoftwarePlatform/MIOpen@2.4.0
+```
+in the file `requirements.txt` (adding a '#' character at the start of each line).
```

Comment:
There is no reason to comment out these lines. `rbuild` should automatically ignore these.